hw09-my_orm/src/main/java/ru/otus/jdbc/templates/JdbcTemplateImpl.java

```diff
@@ -0,0 +1,203 @@
+        this.mapper = new JdbcMapper<>(clazz);
+    }
+
+    @Override
+    public void createTable() throws SQLException {
+        logger.info("Procedure for create table");
+        try(Connection connection = dataSource.getConnection()){
+            executer = new DBExecutorImpl<>(connection);
+            String query = mapper.getCreateTableScript();
+            logger.info("Received query script for create table");
+            executer.formatDB(query);
+            logger.info("Table is created");
+        }
+    }
+
+    @Override
+    public void deleteTable() throws SQLException {
+        logger.info("Delete table procedure");
+        try(Connection connection = dataSource.getConnection()){
+            executer = new DBExecutorImpl<>(connection);
+            String query = mapper.getDropTableScript();
+            logger.info("Received query script for drop table");
+            executer.formatDB(query);
+            logger.info("Table is dropped");
+        }
+    }
+
+    @Override
+    public void create(T objectData) throws SQLException {
+        logger.info("Procedure for insert record");
+        try(Connection connection = dataSource.getConnection()){
+            executer = new DBExecutorImpl<>(connection);
+            String query = mapper.getInsertScript();
+            logger.info("Received query script for insert record");
+            List<Object> params = mapper.getValuesForInsert(objectData);
+            logger.info("Received values("+params.size()+") for insert");
+            Savepoint savepoint = connection.setSavepoint("InsertData");
+            try {
+                long id = executer.insert(query, params);
+                logger.info("Inserted record, id=" + id);
+                connection.commit();
+                logger.info("Committed changes");
+                ReflectionClass.setFieldValue(objectData, mapper.getFieldAutoincrement(), id);
+                logger.info("Seted value to Id field");
+            } catch (SQLException e) {
+                logger.info("Rollback to savepoint. Error: "+e.getMessage());
+                connection.rollback(savepoint);
+            }
+
+        }
+    }
+
+    @Override
+    public void update(T objectData) throws SQLException {
+        logger.info("Procedure for update record");
+        try(Connection connection = dataSource.getConnection()){
+            executer = new DBExecutorImpl<>(connection);
+            String query = mapper.getUpdateScript();
+            logger.info("Received query script for update record");
+            List<Object> params = mapper.getValuesForInsert(objectData);
+            logger.info("Received values("+params.size()+") for update");
+            Savepoint savepoint = connection.setSavepoint("UpdateData");
+            try {
+                Object objectValue = ReflectionClass.getFieldValue(objectData, mapper.getFieldAutoincrement());
+                long id = objectValue != null ? (long) objectValue : -1;
```

Comment:
```suggestion
                long id = objectValue != null ? (long) objectValue : ID_NULL; // или ID_UNKNOWN
```

В именованную константу